tls/test/test_hello_message.py

```diff
@@ -173,6 +180,19 @@ def test_as_bytes_client_hello_cipher_suites(self):
             record.as_bytes()
         assert exc_info.value.args == ('invalid object', 0)
 
+    def test_client_hello_with_server_name_extension(self):
+        """
+        :py:func:`tls.hello_message.ClientHello` parses a packet with a
+        server_name extension
+        """
+        record = ClientHello.from_bytes(self.client_hello_packet_with_server_name_ext)
+        assert len(record.extensions) == 1
+        assert record.extensions[0].type == enums.ExtensionType.SERVER_NAME
+        assert len(record.extensions[0].data) == 1
+        server_name_list = record.extensions[0].data
+        assert server_name_list[0].name_type == enums.NameType.HOST_NAME
+        assert server_name_list[0].name.data == 'localhost'
```

Comment:
I'm not very happy about doing `server_name_list[0].name.data` and would rather `localhost` be `server_name_list[0].name`, i.e. not have the HostName container as a thing. From the RFC (https://tools.ietf.org/html/rfc6066):

```
      struct {
          NameType name_type;
          select (name_type) {
              case host_name: HostName;
          } name;
      } ServerName;

      enum {
          host_name(0), (255)
      } NameType;

      opaque HostName<1..2^16-1>;

      struct {
          ServerName server_name_list<1..2^16-1>
      } ServerNameList;
```